src/main/java/com/teragrep/cfe_39/consumers/kafka/BatchDistributionImpl.java

```diff
@@ -0,0 +1,176 @@
+        long thisTime = Instant.now().toEpochMilli();
+        long ftook = thisTime - lastTimeCalled;
+        topicCounter.setKafkaLatency(ftook);
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER
+                    .debug(
+                            "Fuura searching your batch for <[{}]> with records <{}> and took  <{}> milliseconds. <{}> EPS. ",
+                            topic, batch.size(), (ftook), (batch.size() * 1000L / ftook)
+                    );
+        }
+        long batchBytes = 0L;
+        long start = Instant.now().toEpochMilli();
+        // Starts measuring performance here. Measures how long it takes to process the whole batch.
+
+        // Distribute the records of the batch to a PartitionFileImpl object based on partition from which the record originates from.
+        ListIterator<KafkaRecordImpl> recordOffsetListIterator = batch.listIterator();
+        while (recordOffsetListIterator.hasNext()) {
+            KafkaRecordImpl next = recordOffsetListIterator.next();
+            JsonObject recordOffset = JsonParser.parseString(next.offsetToJSON()).getAsJsonObject();
+            // If the PartitionFileImpl corresponding to the record's partition doesn't exist, create one.
+            if (!partitionFileMap.containsKey(recordOffset.get("partition").getAsString())) {
+                try {
+                    partitionFileMap
+                            .put(recordOffset.get("partition").getAsString(), new PartitionFileImpl(config, recordOffset));
+                }
+                catch (IOException e) {
+                    LOGGER.error("Failed to create new PartitionFileImpl for record <{}>", recordOffset);
+                    throw new RuntimeException(e);
+                }
+            }
+            // Every PartitionFileImpl object will hold responsibility over a single unique file that is related to a single topic partition.
+            PartitionFileImpl recordPartitionFile = partitionFileMap.get(recordOffset.get("partition").getAsString());
+            // Tell PartitionFileImpl to add the current record to the list of records that are going to be added to the file.
+            recordPartitionFile.addRecord(next);
+            batchBytes = batchBytes + next.size(); // metrics
+        }
+
+        // When all records in the current batch have been distributed to different PartitionFileImpl objects successfully, proceed to adding the records to the files for all PartitionFileImpl objects.
+        partitionFileMap.forEach((key, value) -> {
+            try {
+                value.commitRecords();
+            }
+            catch (IOException e) {
+                LOGGER.error("Failed to write the SyslogRecords to PartitionFileImpl <{}> in topic <{}>", key, topic);
+                throw new RuntimeException(e);
+            }
+        });
+
+        // Measures performance of code that is between start and end.
+        long end = Instant.now().toEpochMilli();
```

Comment:
is it justified to use custom metrics instead of dropwizard metrics?

Reply:
Implemented Timer in commit 5f49d37ed639a87998381fb8833d0bca63d29708
Should the metrics usage be refactored further?